src/components/organisms/LocalSearch.tsx

```diff
@@ -0,0 +1,932 @@
+      setSearchValue(event.target.value)
+      if (onSearchQueryChange != null) {
+        onSearchQueryChange(event.target.value)
+      }
+    },
+    [onSearchQueryChange]
+  )
+
+  const focusSearchTextAreaInput = useCallback(
+    (searchValueLength = 0, cursorToEnd = false) => {
+      if (searchTextAreaRef.current == null) {
+        return
+      }
+      searchTextAreaRef.current.focus()
+      if (searchValueLength > 0) {
+        searchTextAreaRef.current.selectionEnd += searchValueLength
+        if (cursorToEnd) {
+          searchTextAreaRef.current.selectionStart =
+            searchTextAreaRef.current.selectionEnd
+        }
+      }
+      setFocusingReplace(false)
+    },
+    []
+  )
+
+  const getSearchRegex = useCallback(
+    (rawSearch, regexSearch: boolean, caseSensitive = caseSensitiveSearch) => {
+      try {
+        const regExp = new RegExp(
+          regexSearch ? rawSearch : escapeRegExp(rawSearch),
+          caseSensitive ? 'gm' : 'gim'
+        )
+        setSearchResultError(() => false)
+        return regExp
+      } catch (err) {
+        // set error for better user acknowledge
+        setSearchResultError(() => true)
+        return null
+      }
+    },
+    [caseSensitiveSearch]
+  )
+
+  const clearMarkers = useCallback(() => {
+    codeMirror.getAllMarks().forEach((mark) => mark.clear())
+    setNumberOfFoundItems(0)
+  }, [codeMirror])
+
+  const scrollToLine = useCallback((editor, lineNumber: number) => {
+    const t = editor.charCoords({ line: lineNumber, ch: 0 }, 'local').top
+    const middleHeight = editor.getScrollerElement().offsetHeight / 2
+    editor.scrollTo(null, t - middleHeight - 5)
+  }, [])
+
+  const focusSearchItem = useCallback(
+    (
+      editor: CodeMirror.EditorFromTextArea,
+      markers: TextMarker[],
+      selectedIndex: number,
+      focusingEditor = false,
+      updateCursor = false
+    ) => {
+      if (selectedIndex >= markers.length || selectedIndex < 0) {
+        console.warn(
+          'Cannot focus editor on selected item.',
+          selectedIndex,
+          markers.length
+        )
+        return
+      }
+
+      const marker = markers[selectedIndex]
+      if (marker == null) {
+        return
+      }
+      const markerPosition: MarkerRange = marker.find() as MarkerRange
+      if (markerPosition == null) {
+        return
+      }
+      const focusLocation = {
+        line: markerPosition.to.line,
+        ch: markerPosition.to.ch,
+      }
+
+      if (!focusingEditor) {
+        scrollToLine(editor, focusLocation.line)
+      }
+      if (updateCursor) {
+        editor.focus()
+        editor.setCursor(focusLocation.line, focusLocation.ch, {
+          scroll: true,
+        })
+        if (focusingEditor && onCursorActivity != null) {
+          onCursorActivity(editor)
+        }
+      }
+    },
+    [scrollToLine, onCursorActivity]
+  )
+
+  const markFoundItems = useCallback(
+    (codeEditor, selectedItemIdx = -1, regExp) => {
```

Comment:
Provide type definitions for arguments of this method

Reply:
I am sorry, codeEditor wasn't typedef because it gave compile error because of the non-existent `getSearchCursor` function. I added type definitions for `regExp` and `codeEditor` now, since selectedItemIdx is inferred by default value